### PR TITLE
Updates to code, pipelines and scripts

### PR DESCRIPTION
**Pull request by xpouyat**
Update some packages, fix pipeline issues, support for main branch, fix admin scripts.
Another PR will be done for the documentation (master to main branch for pipelines)

### CODE REVIEW

**Comment by xpouyat** [comment]
Doc PR is https://github.com/MicrosoftDocs/architecture-center-pr/pull/5185

File: infrastructure/azure-pipelines/templates/steps/terraform-init-steps-template.yml [hunk 1]
```diff
@@ -1,14 +1,14 @@
 parameters:
   provisionStorage: true
-  TerraformVersion: 0.12.20
+  TerraformVersion: 0.12.31
```
Comment (se02035) [inline infrastructure/azure-pipelines/templates/steps/terraform-init-steps-template.yml]: Is there a reason why we don't use a newer versio (>= 1.0.0)?

Comment (xpouyat) [reply infrastructure/azure-pipelines/templates/steps/terraform-init-steps-template.yml]: I tried, but this breaks the pipelines. So there is work to move to > 1.0.0.
It looks like a first step should be to go to 0.13. (https://www.terraform.io/language/upgrade-guides/0-13) but this version also breaks the pipelines.

Using 1.1.4 :
╷
│ Warning: Version constraints inside provider configuration blocks are deprecated
│ 
│   on provider.tf line 9, in provider "azurerm":
│    9:   version = "=2.9.0"
│ 
│ Terraform 0.13 and earlier allowed provider version constraints inside the
│ provider configuration block, but that is now deprecated and will be
│ removed in a future version of Terraform. To silence this warning, move the
│ provider version constraint into the required_providers block.
│ 
│ (and one more similar warning elsewhere)
╵

╷
│ Error: Invalid backend configuration argument
│ 
│ The backend configuration argument "arm_subscription_id" given on the
│ command line is not expected for the selected backend type.
╵

╷
│ Error: Invalid backend configuration argument
│ 
│ The backend configuration argument "arm_tenant_id" given on the command
│ line is not expected for the selected backend type.
╵

╷
│ Error: Invalid backend configuration argument
│ 
│ The backend configuration argument "arm_client_id" given on the command
│ line is not expected for the selected backend type.
╵

╷
│ Error: Invalid backend configuration argument
│ 
│ The backend configuration argument "arm_client_secret" given on the command
│ line is not expected for the selected backend type.
╵

Comment (xpouyat) [reply infrastructure/azure-pipelines/templates/steps/terraform-init-steps-template.yml]: @se02035  I was able to migrated to latest Terraform

**Review by flecoqui — APPROVED** [review]

**Pull request merged**